admin/services/data-access/group.data.service.js

```diff
@@ -19,6 +21,14 @@ groupDataService.sqlFindGroups = async (schoolId) => {
   AND g.school_id=@schoolId
   GROUP BY g.id, g.name
   ORDER BY name ASC`
+  if (atleastOnePresentPupil) {
```

Comment:
I would prefer to keep the data service layer intact from any kind of logic.
Why not provide different data service methods based on what's required from the service layer?
@jon-shipley @GuyHarwood opinions?